src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -762,17 +798,64 @@ export class SearchSource {
     return field;
   }
 
+  public async loadDataViewFields() {
+    const dataView = this.getField('index');
+    if (!dataView || !(dataView instanceof DataViewLazy)) {
+      return;
+    }
+    // eslint-disable-next-line no-console
+    console.log('loadDataViewFields because of DataViewLazy');
+    const request = this.mergeProps(this, { body: {} }, ['query', 'filter']);
+    let fields = dataView.timeFieldName ? [dataView.timeFieldName] : [];
+    const sort = this.getField('sort');
+    if (sort) {
```

Comment:
I was poking through the code to determine if fields referenced in `sort` need to be loaded. Did you figure this out? I dug for a bit but didn't come to a conclusion.

Reply:
yes, if you remove it, it breaks. I think it's because of this: https://github.com/elastic/kibana/blob/844f5150f6539ff4254fd66483ec94b53f66183d/src/plugins/data/common/search/search_source/search_source.ts#L673-L679

which leads to this:
https://github.com/elastic/kibana/blob/40303a88c5735b0b029037db2fb388ea995a07b5/src/plugins/data/common/search/search_source/normalize_sort_request.ts#L44-L58

it seems to be necessary for sorting by a scripted field